lib/types/AbstractUi5Formatter.js

```diff
@@ -0,0 +1,124 @@
+const log = require("@ui5/logger").getLogger("types:AbstractUi5Formatter");
+const path = require("path");
+const fs = require("graceful-fs");
+const AbstractFormatter = require("./AbstractFormatter");
+const {promisify} = require("util");
+const readFile = promisify(fs.readFile);
+
+/**
+ * Base class for formatters that require access to some UI5 specific resources
+ * like manifest.json or pom.xml
+ *
+ * @abstract
+ */
+class AbstractUi5Formatter extends AbstractFormatter {
```

Comment:
Seeing 'Ui5' hurts my eyes...

Reply:
Mine too 🙄

Reply:
Hmm so what about this?

Reply:
Later in some other change 😬

It's private API anyways